geode-management/src/main/java/org/apache/geode/cache/configuration/RegionConfig.java

```diff
@@ -156,7 +159,8 @@
 @XmlType(name = "region-type", namespace = "http://geode.apache.org/schema/cache",
     propOrder = {"regionAttributes", "indexes", "entries", "regionElements", "regions"})
 @Experimental
-public class RegionConfig extends CacheElement implements RestfulEndpoint {
+@JsonIgnoreProperties(value = {"uri"}, allowGetters = true)
+public class RegionConfig extends CacheElement implements RestfulEndpoint<RuntimeRegionConfig> {
```

Comment:
I believe you are trying to use RestfulEndPoint to do a multitask, one is to provide a uri, another is to create a link between the static config and the runtime config. I would prefer you separate these two tasks.

Reply:
Do you have an example in mind where a REST call would have a return type but not a URL, or a URL but not a return type?

Reply:
separated